Tests/Functional/Issue206.php

```diff
@@ -22,5 +22,8 @@ public function testCustomClientStillAvailable()
         // Create a client
         $myCustomClient = new HttpMethodsClient(HttpClientDiscovery::find(), MessageFactoryDiscovery::find());
         $pluginClient = (new PluginClientFactory())->createClient($myCustomClient, []);
+
+        // If we get to this line, no exceptions has been thrown.
```

Comment:
Is this still relevant as we removed the `throw` statement from `ProfileClient`. Maybe you could use `NSA` to check that the `$client` inside `$pluginClient` is a `FlexibleHttpClient`.